src/main/java/org/mifos/connector/bulk/camel/routes/BatchDetailRoute.java

```diff
@@ -0,0 +1,180 @@
+                            Integer.parseInt(exchange.getProperty(FAILED_TRANSACTION_COUNT, String.class)) : 0;
+                    int ongoingTransferCount = exchange.getProperty(ONGOING_TRANSACTION_COUNT)!=null ?
+                            Integer.parseInt(exchange.getProperty(ONGOING_TRANSACTION_COUNT, String.class)) : 0;
+                    int totalTransferCount = exchange.getProperty(TOTAL_TRANSACTION)!=null ?
+                            Integer.parseInt(exchange.getProperty(TOTAL_TRANSACTION, String.class)) : 0;
+
+                    List<Transfer> transfers = batchDetailResponse.getContent();
+                    int completedTransferCountPerPage = 0;
+                    int ongoingTransferCountPerPage = 0;
+                    int failedTransferCountPerPage = 0;
+                    int transferCountPerPage = 0;
+
+                    Map<String, String> previousRequestIdStatusMap;
+                    if(exchange.getProperty(REQUEST_ID_STATUS_MAP)!=null){
+                        previousRequestIdStatusMap = (Map<String, String>) exchange.getProperty(REQUEST_ID_STATUS_MAP);
+                    }
+                    else{
+                        previousRequestIdStatusMap = new HashMap<>();
+                    }
+                    Map<String, String> requestIdStatusMap = new HashMap<>(previousRequestIdStatusMap);
+                    for(Transfer transfer : transfers){
+                        TransferStatus transferStatus = transfer.getStatus();
+                        requestIdStatusMap.put(transfer.getTransactionId(), transferStatus.toString());
+
+                        if(TransferStatus.COMPLETED.equals(transferStatus)){
+                            completedTransferCountPerPage++;
+                        }
+                        else if(TransferStatus.FAILED.equals(transferStatus)){
+                            failedTransferCountPerPage++;
+                        }
+                        else if(TransferStatus.IN_PROGRESS.equals(transferStatus)){
+                            ongoingTransferCountPerPage++;
+                        }
+                        transferCountPerPage++;
+                    }
+
+                    currentTransferCount += transferCountPerPage;
+                    completedTransferCount += completedTransferCountPerPage;
+                    failedTransferCount += failedTransferCountPerPage;
+                    ongoingTransferCount += ongoingTransferCountPerPage;
+
+                    exchange.setProperty(CURRENT_TRANSACTION_COUNT, currentTransferCount);
+                    exchange.setProperty(COMPLETED_TRANSACTION_COUNT, completedTransferCount);
+                    exchange.setProperty(FAILED_TRANSACTION_COUNT, failedTransferCount);
+                    exchange.setProperty(ONGOING_TRANSACTION_COUNT, ongoingTransferCount);
+                    exchange.setProperty(REQUEST_ID_STATUS_MAP, requestIdStatusMap);
```

Comment:
I think this operation is risky, to keep the previous request in here. does this method work synchronously? is it guaranteed that parallel execution will not happen? please explain the logic/ design

Reply:
There is no parallel execution. This route works synchronously based on page number and page size so everytime we are using this worker, we are making a different call.